### PR TITLE
Add seed option to CLI

### LINKED ISSUE #13: Deterministic `random` values
Values generated by the `random` command should actually be deterministically based on the world seed.  We could have some kind of PRNG that gets stored in the `GameState`.  Right now the random values are just generated in the `IO` monad.

### LINKED ISSUE #14: Allow generating worlds from different seeds
Right now, in `Swarm.Game.WorldGen`, some seed values are simply hardcoded.  Instead, especially once we implement #4 , when the user starts a new game they should be able to specify a seed value or have one randomly generated.  Then the seed value should in turn deterministically generate a unique world.

### LINKED ISSUE #90: Tree at base?
**Describe the bug**

I'm not sure if this is intended or not, but upon starting in a new world, `grab` issued at the base will grab a `"tree"`, adding a tree to the base's inventory. The map does not appear to change. Eventually the base tree will regrow and can be `grab`bed again.

**To Reproduce**

Start the game. Enter `grab` at the REPL.

Or just ask `ishere "tree"`.

**Expected behavior**

Either there's no tree at base, or the map shows it ... or maybe it's just an Easter egg.

### PR DESCRIPTION
**Pull request by xsebek**
- adds CLI option `--seed` and propagates it to `testWorld2` (Closes #14)
- moves the base in the tree shade (Closes #90)
- makes the `random` command depend on initial seed (Closes #13)

### CODE REVIEW

**Comment by xsebek** [comment]
Also messes with the `genBiome` settings to get more mountains :sweat_smile: :mountain: 

The main thing to improve is to make `--seed` visible in `--help`, but even better would be to also add biome threshold values loading from some configuration YAML file.

**Comment by xsebek** [comment]
Resolves:
- #90 
- #14

Related (but not solved yet):
- #13

**Comment by xsebek** [comment]
Seed 1 is one of the funny ones :laughing: But think how satisfying it would be to [`drill`](#143) through.

I think these values should be left to the users to tweak and it would help us choose the "best" ones too - we could run:
```sh
while inotifywait -e close_write swarm_conf.yaml; do killall swarm; swarm; done
```

But maybe it is best to make another issue and focus on the `--seed` here :sweat_smile: I will revert the values and try to get the parser right :wink:

**Comment by byorgey** [comment]
Yeah, it makes sense to just add the `--seed` option in one PR and fiddle with world values in another.

> Seed 1 is one of the funny ones laughing But think how satisfying it would be to [`drill`](#143) through.

True. :grin:

**Comment by xsebek** [comment]
The `--help` would now read:
```
Swarm game - pre-alpha version

Usage: swarm [COMMAND | (-s|--seed ARG)]
  To play the game simply run without any command.

Available options:
  -s,--seed ARG            Seed for world generation
  -h,--help                Show this help text

Available commands:
  format                   Format a file
  lsp                      Start the LSP
```

**Comment by xsebek** [comment]
The `random` code may need to be improved - I do not understand Lenses or PRGs that well, and the code uses both. :worried: 
But it seems to give the same numbers every time - `random 6` is 0,0,0,3,4...

File: TUTORIAL.md [hunk 1]
```diff
@@ -307,6 +308,24 @@ you want to reload the definitions.  Eventually, there will be a way
 to both save and load commands, but this is better than nothing for
 now.
 
+World generation
+----------------
+
+If you do not like the starting place of the base, there is a way
+to start somwhere else. *In a different world!*
```
Comment (byorgey) [inline TUTORIAL.md:315]:
```suggestion
to start somewhere else. *In a different world!*
```

File: TUTORIAL.md [hunk 2]
```diff
@@ -307,6 +308,24 @@ you want to reload the definitions.  Eventually, there will be a way
...
+
+```bash
+$ swarm --seed $RANDOM
+```
+
+You can specify the *world seed* leading to a radically different
```
Comment (byorgey) [inline TUTORIAL.md:321]:
```suggestion
You can specify the *world seed* leading to radically different
```

**Review by byorgey — APPROVED** [review]
Looks good to me.  A few minor editing suggestions in the tutorial.

**Comment by byorgey** [comment]
The random code looks fine.  There's probably some clever way to generate a random number and update the `StdGen` in the `GameState` in one line of code, but I don't know what it is off the top of my head, and it would probably be unreadable.

**Pull request merged**